docs/modules/ROOT/pages/explanations/decisions/deployments.adoc

```diff
@@ -0,0 +1,66 @@
+= Decision How to Deploy Compositions and XRDs
+
+== Problem
+
+We need to deploy our XRDs and Compositions for AppCat to OpenShift and Kubernetes clusters.
+Also writing the compositions and XRDs plainly is somewhat a hassle (yaml engineering).
```

Comment:
Not sure if that adds any value. Why is it a hassle? What's the problem with it?

Reply:
Just have a look at the "simple" prototype composition, how easy compositions can escalate: https://github.com/vshn/appcat-service-prototype/blob/master/crossplane/composition.yaml. 

There's almost 400(!) lines of YAML, half of it is just patching things around and taking values from other resources. This is not maintainable as it's hardly understandable...

And then there's the problem of testing them.

Reply:
That's exactly the reason for this point. Should I write a sentence or two about it?

Reply:
yes please

Reply:
Thanks for adding more information, that helps to understand the exact issue :+1: